src/dune_rules/scope.mli

```diff
@@ -13,7 +13,7 @@ val project : t -> Dune_project.t
 (** Return the library database associated to this scope *)
 val libs : t -> Lib.DB.t
 
-val coq_libs : t -> Coq_lib.DB.t
+val coq_libs : t -> Coq_lib.DB.t Lazy.t
```

Comment:
Sorry for my previous misdirection, lazy should not appear here, but internally in scope.

So you have:
```
let coq_libs t = Lazy.force t.coq_db
```

Reply:
I suppose it doesn't make much difference but it is cleaner.

Reply:
It does expose an implementation detail, that should be avoided when possible, you don't want to change all clients when the memoization method changes.